src/librustc_resolve/diagnostics.rs

```diff
@@ -433,6 +452,45 @@ impl<'a> Resolver<'a> {
                 err.span_label(span, "`Self` in type parameter default".to_string());
                 err
             }
+            ResolutionError::UnreachableLabel { name, definition_span, suggestion } => {
+                let mut err = struct_span_err!(
+                    self.session,
+                    span,
+                    E0767,
+                    "use of unreachable label `{}`",
+                    name,
+                );
+
+                err.span_label(definition_span, "unreachable label defined here");
+                err.span_label(span, format!("unreachable label `{}`", name));
+                err.note(
+                    "labels are unreachable through functions, closures, async blocks and modules",
```

Comment:
```suggestion
                    "labels are not reachable through functions, closures, async blocks and modules",
```
Not sure about the wording, sounds a bit weird.

Reply:
I've left the wording as is - it does sound a bit odd and while "not reachable" reads a little bit better in this message, I'd rather it was consistent between all the labels and notes - and I think "not reachable label" would be an odd error message.